operator/v1/types_dns.go

```diff
@@ -101,6 +100,24 @@ type DNSSpec struct {
 	// +optional
 	// +kubebuilder:default=Normal
 	LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+
+	// cache describes the caching configuration that applies to all server blocks listed in the Corefile.
+	// This field allows a cluster admin to optionally configure:
+	// * successTTL which is a duration in seconds for which positive responses should be cached.
+	// * denialTTL which is a duration in seconds for which negative responses should be cached.
+	// +optional
+	Cache *DNSCache `json:"cache,omitempty"`
+}
+
+// DNSCache defines the fields for configuring DNS caching.
+type DNSCache struct {
+	// successTTL is optional and specifies the time (in seconds) that a positive response should be cached.
+	// +optional
```

Comment:
Adding a validation for the minimum and maximum values allowed would be helpful.

Reply:
After a little searching, it looks like there's a precedence to use `metav1.Duration` instead of `time.Duration` or `int64` since `metav1.Duration` has proper marshalling. I think that'll also provide a better UX since user's will be able to configure `3h` instead of `10800`. Then on the backend we can convert the unit to seconds for usage in the Corefile.